src/bnet_client.py

```diff
@@ -0,0 +1,434 @@
+            elif f.key.field == 5:
+                continue
+            elif f.key.field == 6:
+                continue
+            elif f.key.field == 7:
+                continue
+            elif f.key.field == 8:
+                rich_presence = presence_types_pb2.RichPresence()
+                rich_presence.ParseFromString(f.value.message_value)  # e.g. "\rorP\000\025aorp\030\025"
+                game_account_info["rich_presence"] = rich_presence
+            elif f.key.field == 9:
+                continue
+            elif f.key.field == 10:
+                game_account_info["is_away"] = f.value.bool_value
+            elif f.key.field == 11:
+                continue
+
+        future.set_result(game_account_info)
+
+    async def _on_presence__query_account(self, future: asyncio.Future, header, body):
+        if not body:
+            log.error("failed presence:query_response")
+            future.set_result({})
+            return
+
+        response = presence_service_pb2.QueryResponse()
+        response.ParseFromString(body)
+        # log.debug(f"fetched presence:query_response token: {header.token} body: {response}")
+
+        account_info = {}
+        for f in response.field:
+            if f.key.group != 1:
+                continue
+
+            if f.key.field == 1:
+                account_info["full_name"] = f.value.string_value.encode('utf-8')  # e.g. "Firstname Lastname"
+            elif f.key.field == 3:
+                if "game_accounts" not in account_info:
+                    account_info["game_accounts"] = []
+                account_info["game_accounts"].append({"id": f.value.entityid_value})  # e.g. high: 144115197778542960 low: 131237370
+            elif f.key.field == 4:
+                account_info["battle_tag"] = f.value.string_value  # e.g. "Username#1234"
+            elif f.key.field == 6:
+                account_info["last_online"] = datetime.fromtimestamp(f.value.int_value/1000/1000)  # e.g. 1584194739362351 (microseconds timestamp)
+            elif f.key.field == 7:
+                account_info["is_away"] = f.value.bool_value
+            elif f.key.field == 8:
+                account_info["away_time"] = f.value.int_value  # e.g. 1583607638026991
+            elif f.key.field == 11:
+                account_info["is_busy"] = f.value.bool_value
+
+        future.set_result(account_info)
+
+    async def _on_friends__subscribe_to_friends(self, future: asyncio.Future, header, body):
+        if not body:
+            log.error("failed friends:subscribe_to_friends_response")
+            future.set_result([])
+            return
+
+        response = friends_service_pb2.SubscribeToFriendsResponse()
+        response.ParseFromString(body)
+        # log.debug(f"fetched friends:subscribe_to_friends_response token: {header.token} body: {response}")
+        # log.debug(f"fetched friends:subscribe_to_friends_response token: {header.token}")
```

Comment:
redundant

Reply:
and what about module level logging like 

```
log = logging.getLogger(__name__)
log.setLevel = logging.INFO
```

Reply:
Thx for the suggestion, I have done this.